### DISCUSSION TITLE
Force path-style property

### DISCUSSION THREAD

**Discussion by mpribela** [awspring/spring-cloud-aws#857] [General]
Hi, is there a way to set 'force path style' via properties? If not, do you plan to add it?
Thank you

**Comment by MatejNedic** [comment]
Hey @mpribela , could you be more specific, I don't understand the question. Could you please provide a example of what you mean?

**Comment by mpribela** [comment]
When I define my own S3AsyncClient, I can set forcePathStyle to true. But by default, the provided S3AsyncClient doesn't have this option set and I thought that there could be a configuration property such as `spring.cloud.aws.forcePathStyle'.

Right now I am facing an issue and setting up this property is what it makes work.

I am running localstack in docker container with exposed port 4566.
When I try for example:

`s3AsyncClient.createBucket(builder -> builder.bucket("test")).join();`

I get an error: 
> software.amazon.awssdk.core.exception.SdkClientException: Failed to send the request: Host name was invalid for dns resolution.

in the application.properties: 
```
spring.cloud.aws.credentials.access-key=test
spring.cloud.aws.credentials.secret-key=test
spring.cloud.aws.region.static=us-east-1
spring.cloud.aws.endpoint=http://localhost:4566
spring.cloud.aws.s3.path-style-access-enabled=true
```
when I try:
```
S3AsyncClient.crtBuilder()
                .credentialsProvider(StaticCredentialsProvider.create(AwsBasicCredentials.create("test", "test")))
                .forcePathStyle(true)
                .endpointOverride(URI.create("http://localhost:4566"))
                .region(Region.US_EAST_1)
                .build().createBucket(builder -> builder.bucket("test")).join();
```
the bucket is created. I am not sure if forcePathStyle to true is the solution or a side effect of why it works. So I wanted to override only this property.
